### PR TITLE
Refactor authentication and gateways

### PR DESCRIPTION
**Pull request by 1239uth**

### CODE REVIEW

**Review by nxabdullah — CHANGES_REQUESTED** [review]
Explain why you removed the View interface, this is a violation of MVP.

**Comment by 1239uth** [comment]
> Explain why you removed the View interface, this is a violation of MVP.

The presenter is not accessing the activity directly so it should be fine. It's still using an interface, it's just a different one. The reason it's not nested inside the class is because the same interface is used my multiple classes so its in its own file. Its called OpensActivityWithProfile. (open to suggestions on a shorter name...)

File: app/src/main/java/com/example/fitappa/Authentication/DatabaseConstants.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class DatabaseConstants {
+
+    /**
+     * Get the collection name for users
+     *
+     * @return String representing collection name for users
+     */
+    public String getUsersCollection() {
+        return "users";
```
Comment (sourenrex) [inline app/src/main/java/com/example/fitappa/Authentication/DatabaseConstants.java:14]: arent constants supposed to be all capital?

Comment (1239uth) [reply app/src/main/java/com/example/fitappa/Authentication/DatabaseConstants.java:14]: constant variable names are supposed to be capital not string values. unless you mean the method name should be capital?

**Comment by nxabdullah** [comment]
> > Explain why you removed the View interface, this is a violation of MVP.
> 
> 
> 
> The presenter is not accessing the activity directly so it should be fine. It's still using an interface, it's just a different one. The reason it's not nested inside the class is because the same interface is used my multiple classes so its in its own file. Its called OpensActivityWithProfile. (open to suggestions on a shorter name...)

I don't know if this is a good idea. It seems like, with this approach we need to create an interface every time we want a new method. As per the MVP architecture we are using, the point of the interface is that it allows the presenter to call methods inside the view. Since each presenter corresponds to one view, we should have one interface per presenter. In the short term, your approach works because there is only one method right now, but once there are more we will have to clog our directory with interfaces leading to badly designed architecture. 

These are my two cents, the decision is up to you

**Review by nxabdullah — APPROVED** [review]

**Comment by 1239uth** [comment]
> > > Explain why you removed the View interface, this is a violation of MVP.
> > 
> > 
> > The presenter is not accessing the activity directly so it should be fine. It's still using an interface, it's just a different one. The reason it's not nested inside the class is because the same interface is used my multiple classes so its in its own file. Its called OpensActivityWithProfile. (open to suggestions on a shorter name...)
> 
> I don't know if this is a good idea. It seems like, with this approach we need to create an interface every time we want a new method. As per the MVP architecture we are using, the point of the interface is that it allows the presenter to call methods inside the view. Since each presenter corresponds to one view, we should have one interface per presenter. In the short term, your approach works because there is only one method right now, but once there are more we will have to clog our directory with interfaces leading to badly designed architecture.
> 
> These are my two cents, the decision is up to you
> 
> Edit: With one interface, it is fine but if we start repeating the process it can become bad, since no other methods are needed in the presenter ATM, i think it's fine to omit it right now. But just be aware, when profile does get complex, the view interface will have to come back

Yea its not being used right now so theres no point in keeping it. Good point though

**Pull request merged**